core/src/main/java/com/uber/m3/tally/TimerImpl.java

```diff
@@ -94,7 +94,7 @@ public void recordStopwatch(long stopwatchStart) {
     }
 
     static class Values {
-        private ReadWriteLock rwlock = new ReentrantReadWriteLock();
+        private final ReadWriteLock rwlock = new ReentrantReadWriteLock();
```

Comment:
why use a rwlock here v synchronise in other places?

Reply:
`synchronized` is used when I would only use one (write) lock. We're making use of ConcurrentHashMap's ability to read values in a thread-safe manner without locks. This rwlock is to protect ArrayList accesses (there isn't a similar Concurrent* version) and having separate read/write locks allow for higher throughput.

Reply:
makes sense, could you put a note in the code explaining that intent